webui/react/src/pages/ExperimentDetails/CodeViewer/CodeViewer.tsx

```diff
@@ -385,6 +404,30 @@ const CodeViewer: React.FC<Props> = ({
     ) configDownloadButton.current.click();
   }, [ downloadInfo ]);
 
+  useEffect(
+    () => {
+      if (settings.fileName && (activeFile?.title !== settings.fileName)) {
+        const fileNode = treeData.find((node) => node.title === settings.fileName);
```

Comment:
this doesn't work for files that are not in the root directory. 

we should probably `useSettings` for the the path/key as opposed to the title.

should be able to do something like this to avoid duplicating things.

```
useEffect(() => {
    handleSelectFile({path: settings.filePath, title: <extract title from path>} as DataNode})
, [settings.filePath, handleSelectFile])
```